src/main/java/com/exactpro/th2/taskutils/FutureTracker.java

```diff
@@ -98,27 +209,81 @@ public void awaitRemaining (long timeoutMillis) {
                     }
                 }
             } catch (ExecutionException | TimeoutException e) {
+                // do nothing
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
             } catch (Exception e) {
                 throw new RuntimeException(e);
             }
         }
+        if (Thread.currentThread().isInterrupted()) {
+            throw new InterruptedException();
+        }
```

Comment:
question: Do we need to throw the `InterruptedException`? `get` method will throw the exception itself. There is no other blocking operations